packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/mira/MiraController.java

```diff
@@ -269,47 +271,71 @@ public ResponseEntity<JsonNode> generateModelLatex(@RequestBody final JsonNode m
 		}
 	)
 	public ResponseEntity<JsonNode> latexToAMR(@RequestBody final String latex) {
-		// create request:
-		final TaskRequest req = new TaskRequest();
-		req.setType(TaskType.MIRA);
+		////////////////////////////////////////////////////////////////////////////////
+		// 1. Convert latex string to python sympy code string
+		//
+		// Note this is a gollm string => string task
+		////////////////////////////////////////////////////////////////////////////////
+		final TaskRequest latexToSympyRequest = new TaskRequest();
```

Comment:
Why not use the `CompoundTask` for that purpose?

Reply:
It looks like `CompoundTask` runs tasks independently, it doesn't really have the notion of chaining tasks together (eg what is the handoff between taskA and taskB) or some kind of computation graph.